### PR TITLE
feat: Move DB connection to azure key vault

### PR DESCRIPTION
**Pull request by Yakitrak**
## Describe your changes

Ticket: https://pins-ds.atlassian.net/browse/ASB-1924

Removes DB connection from variables and adds to azure key vault

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] My code changes do not include any hardcoded secrets or passwords
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] My Tech Lead is aware of any infrastructure changes that will cost money

### CODE REVIEW

File: app/components/applications-app-services/variables.tf [hunk 1]
```diff
@@ -39,11 +39,6 @@ variable "applications_sql_server_connection_string" {
   type        = string
 }
 
-variable "applications_sql_server_connection_string_jbdc" {
-  description = "Connection string for applications service sql server in JBDC format"
-  type        = string
-}
-
```
Comment (JamesP-KC) [inline app/components/applications-app-services/variables.tf:46]: I think you can also remove `variable "applications_sql_server_connection_string"`. I can't see it being used anywhere within `app/components/applications-app-services`

File: app/stacks/uk-south/applications-service/app-service.tf [hunk 2]
```diff
@@ -10,8 +10,7 @@ module "app_services" {
   app_service_private_dns_zone_id                                 = data.azurerm_private_dns_zone.app_service.id
   applications_service_public_url                                 = var.applications_service_public_url
   applications_service_redis_connection_string_secret_name        = azurerm_key_vault_secret.redis_cache_connection_string[count.index].name
-  applications_sql_server_connection_string                       = local.sql_connection_string
-  applications_sql_server_connection_string_jbdc                  = local.sql_jbdc_connection_string
+  applications_sql_server_connection_string                       = azurerm_key_vault_secret.applications_sql_server_connection_string
```
Comment (JamesP-KC) [inline app/stacks/uk-south/applications-service/app-service.tf:13]: I think this variable can be removed. As above, I don't think the `applications-app-services` component is actually using it.

File: app/components/applications-app-services/variables.tf (hunk 1 above)
Comment (JamesP-KC) [reply app/components/applications-app-services/variables.tf:46]: TFLint is picking this up actually:

`TFLint in app/components/applications-app-services/:
1 issue(s) found:

Warning: [Fixable] variable "applications_sql_server_connection_string" is declared but not used (terraform_unused_declarations)

  on variables.tf line 37:
  37: variable "applications_sql_server_connection_string" {

Reference: https://github.com/terraform-linters/tflint-ruleset-terraform/blob/v0.4.0/docs/rules/terraform_unused_declarations.md
`

File: app/stacks/uk-south/applications-service/app-service.tf (hunk 2 above)
Comment (Yakitrak) [reply app/stacks/uk-south/applications-service/app-service.tf:13]: Will remove but do you mind explaining please? Isn't this used in our application service?

**Review by JamesP-KC — APPROVED** [review]

**Pull request merged**